dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/storage/GitHubJsonStore.java

```diff
@@ -43,31 +49,51 @@ public void updateStore(String img, String tag) throws IOException {
         updateStoreOnGithub(storeRepo, Constants.STORE_JSON_FILE, img, tag);
     }
 
-    protected void updateStoreOnGithub(GHRepository repo, String path, String img, String tag) throws IOException {
+    private void initializeTagStoreIfRequired(GHRepository repo, String path) throws IOException {
         try {
             repo.getFileContent(path);
-        } catch (IOException e) {
+        } catch (FileNotFoundException e) {
+            log.info("Image tag store {} not found. Creating a new image tag store", path);
             repo.createContent().content("").message("initializing store").path(path).commit();
+        } catch (IOException ex) {
+            if (ex.getMessage().contains("too_large")) {
+                log.info("Image tag store {} is already initialized and have size more than 1 MB"
+                        , path);
+            } else {
+                throw ex;
+            }
         }
+    }
 
-        String latestCommit = repo.getBranches().get(repo.getDefaultBranch()).getSHA1();
-        log.info("Loading image store at commit {}", latestCommit);
-        GHContent content = repo.getFileContent(path, latestCommit);
-
-        if (content.isFile()) {
-            JsonElement json;
-            try (InputStream stream = content.read(); InputStreamReader streamR = new InputStreamReader(stream)) {
-                try {
-                    json = JsonParser.parseReader(streamR);
-                } catch (JsonParseException e) {
-                    log.warn("Not a JSON format store. Clearing and rewriting as JSON...");
-                    json = JsonNull.INSTANCE;
-                }
-            }
-            String jsonOutput = getAndModifyJsonString(json, img, tag);
-            content.update(jsonOutput,
-                    String.format("Updated image %s with tag %s.\n@rev none@", img, tag), repo.getDefaultBranch());
+    protected void updateStoreOnGithub(GHRepository repo, String path, String img, String tag) throws IOException {
+        initializeTagStoreIfRequired(repo, path);
+
+        GHCommit commit = repo.getCommit(repo.getDefaultBranch());
+        GHBlob blob = commit.getTree().getEntry(path).asBlob();
+        JsonElement json = null;
+
+        try {
+            String text = IOUtils.toString(blob.read(), StandardCharsets.UTF_8.name());
```

Comment:
I rather not add an extra dependency on IOUtils. You should be able to use the language builtin here:
```
new String(blob.read().readAllBytes(), StandardCharsets.UTF_8)
```

Reply:
Also, your blob.read() returns a Closable, so you need to add the try with resources back. I recommend something like this: 
```
String text;
try (InputStream stream = blob.read()) {
    text = new String(blob.read().readAllBytes(), StandardCharsets.UTF_8);
}

```